Windows build error: 'executor_type': is not a member of 'mock_ssl_socket'

When following the 'How to Build' procedures for Windows the below compilation errors are encountered.

A basic observation is that mock_ssl_socket is not a stream template like asio::ssl::stream which is expected by the read and write template.

Perhaps this issue should be posted to the asio repository?

When able, please advise.

COMPILE ERRORS
\NuRaft\asio\asio\include\asio/impl/read.hpp(504): error C2039: 'executor_type': is not a member of 'mock_ssl_socket'
\NuRaft\asio\asio\include\asio/impl/read.hpp(504): error C3646: 'executor_type': unknown override specifier
\NuRaft\asio\asio\include\asio/impl/write.hpp(469): error C2039: 'executor_type': is not a member of 'mock_ssl_socket'
\NuRaft\asio\asio\include\asio/impl/write.hpp(469): error C3646: 'executor_type': unknown override specifier


ASIO_SERVICE.CXX
#ifdef SSL_LIBRARY_NOT_FOUND
    #include "mock_ssl.hxx"
    using ssl_socket = mock_ssl_socket;
    using ssl_context = mock_ssl_context;
#else
    #include "asio/ssl.hpp"
    using ssl_socket = asio::ssl::stream<asio::ip::tcp::socket&>;
    using ssl_context = asio::ssl::context;
#endif

DETAIL EXCERPT
|Tue 07/27/2021|11:31|D:\B\NuRaft\build> nmake

Microsoft (R) Program Maintenance Utility Version 14.29.30038.1
Copyright (C) Microsoft Corporation.  All rights reserved.

[  1%] Building CXX object CMakeFiles/RAFT_CORE_OBJ.dir/src/asio_service.cxx.obj
asio_service.cxx
Please define _WIN32_WINNT or _WIN32_WINDOWS appropriately. For example:
- add -D_WIN32_WINNT=0x0601 to the compiler command line; or
- add _WIN32_WINNT=0x0601 to your project's Preprocessor Definitions.
Assuming _WIN32_WINNT=0x0601 (i.e. Windows 7 target).
D:\B\NuRaft\asio\asio\include\asio/impl/read.hpp(504): error C2039: 'executor_type': is not a member of 'mock_ssl_socket'
...

--- a/src/mock_ssl.hxx
+++ b/src/mock_ssl.hxx
@@ -28,8 +28,8 @@ public:
 
 class mock_ssl_socket {
 public:
-    using executor_type = int;
     using lowest_layer_type = asio::ip::tcp::socket;
+    using executor_type = lowest_layer_type::executor_type;
 
     mock_ssl_socket(asio::ip::tcp::socket& tcp_socket,
                     mock_ssl_context& context)

